python/lib/core/dmod/core/meta_data.py

```diff
@@ -622,12 +797,17 @@ def handle_type_map(t):
 
     @root_validator()
     def validate_sufficient_restrictions(cls, values):
+        data_format = values.get("data_format")
+        if data_format == DataFormat.EMPTY or data_format == DataFormat.GENERIC:
```

Comment:
I would add a comment here that it is permissible for an `EMPTY` or `GENERIC` to have restrictions.

Reply:
Related to this, we should document dataset invariants at some point. So almost like a retrospective design doc. Seems like something we could put together in a DMOD team brown bag that deep dives into the concepts of a DMOD dataset.